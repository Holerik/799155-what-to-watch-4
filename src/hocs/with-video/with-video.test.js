// with-video.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withVideo from './with-video.jsx';


const MockComponent = (props) => {
  const {children} = props;
  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withVideo(MockComponent);

describe(`withVideo tests`, () => {
  it(`Should correctly render VideoPlayer`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        isPlaying={true}
        isMuted={true}
        src={``}
        poster={``}
        width={200}
      />), {
      createNodeMock() {
        return {};
      }
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
