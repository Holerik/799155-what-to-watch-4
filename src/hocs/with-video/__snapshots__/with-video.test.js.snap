// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideo tests Should correctly render VideoPlayer 1`] = `
<div>
  <video
    className="player__video"
    muted={true}
    poster=""
    width={200}
  >
    <source
      src=""
    />
  </video>
</div>
`;
